SCSH_3C_iOS/SCSH_3C_iOS/Repository/Mapper/DistributorMapper.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  DistributorMapper.swift
+//  SCSH_3C_iOS
+//
+//  Created by 辜敬閎 on 2023/12/6.
+//
+
+import Foundation
+import Persistence
+import Networking
+import CoreData
+
+protocol DistributorMapperType {
+    func transform(networkDistributors distributors: [Networking.Distributor], context: NSManagedObjectContext) -> [Persistence.Distributor]
+    func transform(persistenceDistributors distributors: [Persistence.Distributor]) -> [Distributor]
+    
+    func transform(networkDistributorLikes likes: [Networking.DistributorLike], context: NSManagedObjectContext) -> [Persistence.Distributor_Like]
+}
+
+class DistributorMapper: DistributorMapperType {
+    
+    private let encoder = JSONEncoder()
+    private let decoder = JSONDecoder()
+    
+    func transform(networkDistributors distributors: [Networking.Distributor], context: NSManagedObjectContext) -> [Persistence.Distributor] {
+        distributors.map { distributor in
+            let persistenceDistributor = Persistence.Distributor(context: context)
+            persistenceDistributor.id = Int64(distributor.id)
+            persistenceDistributor.name = distributor.name
+            persistenceDistributor.detail = distributor.description
+            persistenceDistributor.brand_image = URL(string: distributor.brandImage)
+            persistenceDistributor.products = transform(networkDistributorProducts: distributor.products)
+            
+            return persistenceDistributor
+        }
+    }
+    
+    func transform(persistenceDistributors distributors: [Persistence.Distributor]) -> [Distributor] {
+        var result: [Distributor] = []
+        
+        for distributor in distributors {
+            
+            var products: [Distributor.Product] = []
+            
+            if let productsData = distributor.products {
+                products = (try? self.decode(distributorProductsData: productsData)) ?? []
+            }
+            
+            result.append(Distributor(id: Int(distributor.id),
+                                      name: distributor.name ?? "",
```

Comment:
Maybe we need to ignore the Distributor without a name, it doesn't not make sense for user.

Reply:
1. In the GET /distributors endpoint,  the `name` column is defined as a non-optional type.
2. Due to the Null safety considerations, all properties in the CoreData entity are treated as optional, even though marked as required (unless the entity is customized). 

As the factors shown above, the null situations are barely gonna happen unless exceptions occur, that's annoying and makes the code obscure 😵.

Solutions:
1. Implement logic to ignore null values within the SCSH_3C_iOS.
2. Customize the entity in the Persistence to fix this issue.

I'm currently leaning toward option 2, Do you have any insights or suggestions?

Reply:
> Customize the entity in the `Persistence` module to fix this issue

It sounds good. But I'm not sure if it requires adding a lot of corresponding code. 

I think this optimization can be deferred as well.